lib/pos_handshake_controller.go

```diff
@@ -0,0 +1,173 @@
+package lib
+
+import (
+	"github.com/decred/dcrd/lru"
+	"github.com/deso-protocol/core/bls"
+	"github.com/golang/glog"
+	"math"
+)
+
+// TODO: Replace this
+func getActiveValidators() map[bls.PublicKey]*ValidatorEntry {
+	// TODO: replace with a getter to retrieve all active validators.
+	activeValidators := []*ValidatorEntry{}
+	allValidatorsMap := make(map[bls.PublicKey]*ValidatorEntry)
+	for _, validator := range activeValidators {
+		pk := validator.VotingPublicKey
+		if pk == nil {
+			continue
+		}
+		allValidatorsMap[*pk] = validator
+	}
+
+	return allValidatorsMap
+}
+
+type HandshakeController struct {
+	rnManager              *RemoteNodeManager
+	usedNonces             lru.Cache
+	protocolOnProofOfStake func() bool                              // TODO
+	getActiveValidators    func() map[bls.PublicKey]*ValidatorEntry // TODO
+}
+
+func NewHandshakeController(rnManager *RemoteNodeManager) *HandshakeController {
+
+	vm := &HandshakeController{
+		rnManager:              rnManager,
+		usedNonces:             lru.NewCache(1000),
+		protocolOnProofOfStake: func() bool { return false },
```

Comment:
See comment below in the `handlePoSHandshakePeerMessage` function. It seems like we can do away with this function